src/exposurelog/routers/add_message.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import annotations
+
+__all__ = ["add_message"]
+
+import asyncio
+import typing
+
+import astropy.time
+import astropy.units as u
+import fastapi
+import lsst.daf.butler
+import sqlalchemy as sa
+
+from ..message import ExposureFlag, Message
+from ..shared_state import SharedState, get_shared_state
+
+router = fastapi.APIRouter()
+
+
+@router.get("/add_message/", response_model=Message)
```

Comment:
OK. I changed all commands except find_messages to POST. I'm certainly willing to use the other kinds of requests, but I'd rather keep it simple and it seems a common pattern to only use GET or POST for a web API.

Reply:
Yes, POST is fine. GET/POST Is very common and will work without any trouble and address the security concern.

For the record, just FYI, part of the REST "specification" to the extent that there is one is that each object the app manages should have a unique URL and the REST API is supposed to use the full set of HTTP verbs to change those objects. So, for example, a full REST pattern for this application would look something like:

- POST to `/exposurelog/messages` to create a new message, returns a redirect to `/exposurelog/messages/1`
- GET `/exposurelog/messages/1` to read message 1
- DELETE `/exposurelog/messages/1` to delete message 1
- PUT `/exposurelog/messages/1` to replace message 1 with new content
- PATCH `/exposurelog/messages/1` to change specific fields of message 1
- GET `/exposurelog/messages?param=...&other=...` to search for messages matching some criteria, which would return URLs to those messages

It's perfectly fine to not follow the REST API model and sometimes it's not a good fit for a particular application. In this case it would mean a significant restructuring of the URL structure of your app to no real purpose. But that pattern is technically what REST is intended to mean.

Reply:
Thanks. I guess the id and site_id fields could be combined to create a unique URL, e.g. `exposurelog/messages/1_summit`. I will say that I am far more interested in making this service use OpenAPI and standard requests (e.g. GET/POST) than I am in making it meet the full REST specification. I'm not sure what purpose the latter serves.

I am confused why `DELETE /exposurelog/messages/1` is safe. It requires no form parameters, so it has the security concern that you expressed about GET: one can send the URL around.

I'm not thrilled with making users delete one message at a time, but I can certainly change that if necessary. I don't know how common delete will be, but it's just as easy to take a list of IDs as it is to take a single ID.

Anyway, this explains why the fastAPI docs make such a big deal out of URL parameters. I was wondering!